src/models/gun.rs

```diff
@@ -0,0 +1,73 @@
+use std::time::{Instant};
+
+const DEFAULT_NATURAL_COOL_DOWN_RATE: f32 = 0.2;
+const DEFAULT_OVERHEAT_COOL_DOWN_RATE: f32 = 0.40;
+const DEFAULT_MAXIMUM_CAPACITY: f32 = 1.0;
+const DEFAULT_MINIMUM_CAPACITY: f32 = 0.0;
+const DEFAULT_HEAT_UP: f32 = 0.05;
+
+pub struct Gun {
+    pub minimum_capacity: f32,
+    pub maximum_capacity: f32,
+    pub temperature: f32,
+    natural_cool_down_rate: f32,
+    overheated: bool,
+    overheat_cool_down_rate: f32,
+    last_update: Instant
+}
```

Comment:
Since we are already using the `DEFAULT_*` constants, I think we could simplify further and remove the following fields from the struct:

* `minimum_capacity`
* `maximum_capacity`
* `natural_cool_down_rate`
* `overheat_cool_down_rate`

I also think we should get rid of `last_update`, but I will comment below on that

Reply:
I'd rather keep these in the default object than keep them as consts.